runtime/backtrace_byt.c

```diff
@@ -275,34 +275,38 @@ code_t caml_next_frame_pointer(value ** sp, value ** trsp)
   return NULL;
 }
 
-intnat caml_current_callstack_size(intnat max_frames)
+#define Default_callstack_size 32
+intnat caml_collect_current_callstack(value** pbuffer, intnat* plen,
+                                      intnat max_frames, int alloc_idx)
 {
-  intnat trace_size;
-  value * sp = Caml_state->extern_sp;
+  value * sp =  Caml_state->extern_sp;
   value * trsp = Caml_state->trapsp;
+  intnat trace_pos = 0, trace_len = *plen;
+  value* trace = *pbuffer;
```

Comment:
Caching `trace_len` and `trace` makes the code harder to read. Have you actually observed some performance benefit doing so?

Reply:
The locals are there for correctness in the reallocation case:
```
      trace_len *= 2;
      trace = caml_stat_resize_noexc(trace, trace_len * sizeof(value));
      if (trace == NULL) break;
      *pbuffer = trace;
      *plen = trace_len;
```
If `caml_stat_resize_noexc` fails, the size has not changed and `*plen` should not be updated.

Reply:
Right. But then I would have only a local at the allocation site.

Reply:
I think it's cleaner to just introduce the locals where needed, I'll push a new version.